### PR TITLE
Vfat 01

### PR DESCRIPTION
**Pull request by andrewbird**
It's been a while since I wrote this and I don't really have the time now to properly understand the existing code that it's modifying. The tests I added still pass, but I'd appreciate somebody to look it over closely.

### CODE REVIEW

**Comment by stsp** [comment]
Why do we need to convert
sfn to lfn in find_file()? So far
it was only fixing up the letter-case
in path.

**Comment by andrewbird** [comment]
Perhaps there's a better place to do it, but somewhere the vfat mangled sfn needs to change to lfn. Note that the vfat sfn mangling method is different from those that dosemu generates and caches for regular unix filesystems.

**Comment by stsp** [comment]
But could you please describe
the scenario when/why this is needed?

**Comment by andrewbird** [comment]
Mmm indeed, I dropped that part of the PR and tests still pass. I wonder why I thought it necessary to do it. Repushed.

File: src/dosext/mfs/lfn.c [hunk 1]
```diff
@@ -102,26 +102,26 @@ void close_dirhandles(unsigned psp)
 
 static int vfat_search(char *dest, char *src, char *path, int alias)
 {
-	struct mfs_dir *dir = dos_opendir(path);
```
Comment (stsp) [inline src/dosext/mfs/lfn.c:105]: I won't say that 2-spaces
indent is an "official one"
in dosemu2. Its definitely
inherited from dosemu1
though, and if you like it -
oh well.

Comment (andrewbird) [reply src/dosext/mfs/lfn.c:105]: But I thought we'd agreed on it here #514 and here https://github.com/dosemu2/dosemu2/blob/devel/CONTRIBUTING.md ?

Comment (stsp) [reply src/dosext/mfs/lfn.c:105]: Ah, maybe. :)
In any case, its not always
followed. Its just that
reformatting the entire code
base to something else would
be stupid.
Anyway, I am not against your patch.

**Comment by andrewbird** [comment]
So just for completeness I tried Freecom 0.84 (from FreeDOS 1.2) on vfat mounted directory

~~~
$ ls -l /mnt/dosemu
total 3
-rwxr-xr-x 1 ajb sphere   9 Jun 13 15:47  MixedCaseFilename.ext
drwxr-xr-x 2 ajb sphere 512 Jun 13 15:47 'Program Files'
-rwxr-xr-x 1 ajb sphere   9 Jun 13 15:47 'space embedded filename.txt'
-rwxr-xr-x 1 ajb sphere   9 Jun 13 15:47  verylongfilename2.txt
-rwxr-xr-x 1 ajb sphere   9 Jun 13 15:47  verylongfilename.txt
~~~

~~~
C:\>lredir x: /mnt/dosemu
X: = \\LINUX\FS/mnt/dosemu attrib = READ/WRITE
C:\>dir x: /lfn
 Volume in drive X is  MNT DOSEMU
 Directory of X:\
 
PROGRA~1             <DIR>  06-13-21  3:47p Program Files
VERYLO~1 TXT             9  06-13-21  3:47p verylongfilename.txt
VERYLO~2 TXT             9  06-13-21  3:47p verylongfilename2.txt
SPACEE~1 TXT             9  06-13-21  3:47p space embedded filename.txt
MIXEDC~1 EXT             9  06-13-21  3:47p MixedCaseFilename.ext
         4 file(s)             36 bytes
         1 dir(s)       1,455,104 bytes free 
~~~

So that looks good to me.

**Comment by andrewbird** [comment]
BTW I just added on a little tweak to reduce the number of DOSes tested by Travis(now they've fixed the repo) so that it won't exceed their maximum test time limit.

**Comment by stsp** [comment]
Thanks!
Another important fix.

**Pull request merged**